### ISSUE TITLE
URL preview horribly mangled because of autolinker

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#4453]
Twitter post: https://twitter.com/DeadSuperHero/status/963565078766219264
Friendica post: https://friendica.mrpetovan.com/display/a6551d7191250acb62b830e006705dafe5a81db7

Source object:
````json
{
	"created_at": "Wed Feb 14 00:06:31 +0000 2018",
	"id": 963565078766219264,
	"id_str": "963565078766219264",
	"full_text": "For obvious reasons, this is kind of a funny status: https:\/\/t.co\/1C3GseycsZ\n\"Published via Socialhome via https:\/\/t.co\/LFJtewY7f9 via Friendica via Diaspora\"",
	"truncated": false,
	"display_text_range": [0, 158],
	"entities": {
		"hashtags": [],
		"symbols": [],
		"user_mentions": [],
		"urls": [{
			"url": "https:\/\/t.co\/1C3GseycsZ",
			"expanded_url": "https:\/\/joindiaspora.com\/posts\/11154632",
			"display_url": "joindiaspora.com\/posts\/11154632",
			"indices": [53, 76]
		}, {
			"url": "https:\/\/t.co\/LFJtewY7f9",
			"expanded_url": "http:\/\/Pump.io",
			"display_url": "Pump.io",
			"indices": [107, 130]
		}]
	},
	"source": "<a href=\"http:\/\/twitter.com\" rel=\"nofollow\">Twitter Web Client<\/a>",
	"in_reply_to_status_id": null,
	"in_reply_to_status_id_str": null,
	"in_reply_to_user_id": null,
	"in_reply_to_user_id_str": null,
	"in_reply_to_screen_name": null,
	"user": {
		"id": 71139907,
		"id_str": "71139907",
		"name": "Sean Tilley",
		"screen_name": "DeadSuperHero",
		"location": "",
		"description": "Customer Success @getsentry. Worked @joindiaspora and @BackerKit in the past. I like hard puzzles, trippy art, adventure games, weird movies, GNU\/Linux.",
		"url": "https:\/\/t.co\/82BqiIG4JR",
		"entities": {
			"url": {
				"urls": [{
					"url": "https:\/\/t.co\/82BqiIG4JR",
					"expanded_url": "http:\/\/deadsuperhero.com",
					"display_url": "deadsuperhero.com",
					"indices": [0, 23]
				}]
			},
			"description": {
				"urls": []
			}
		},
		"protected": false,
		"followers_count": 978,
		"friends_count": 2082,
		"listed_count": 52,
		"created_at": "Thu Sep 03 01:47:55 +0000 2009",
		"favourites_count": 4528,
		"utc_offset": -28800,
		"time_zone": "Pacific Time (US & Canada)",
		"geo_enabled": true,
		"verified": false,
		"statuses_count": 7700,
		"lang": "en",
		"contributors_enabled": false,
		"is_translator": false,
		"is_translation_enabled": false,
		"profile_background_color": "000000",
		"profile_background_image_url": "http:\/\/pbs.twimg.com\/profile_background_images\/378800000083763734\/9e7ef91d9955c204e25b6abdc01b0b77.jpeg",
		"profile_background_image_url_https": "https:\/\/pbs.twimg.com\/profile_background_images\/378800000083763734\/9e7ef91d9955c204e25b6abdc01b0b77.jpeg",
		"profile_background_tile": false,
		"profile_image_url": "http:\/\/pbs.twimg.com\/profile_images\/914215335577841664\/PF8XkG2v_normal.jpg",
		"profile_image_url_https": "https:\/\/pbs.twimg.com\/profile_images\/914215335577841664\/PF8XkG2v_normal.jpg",
		"profile_banner_url": "https:\/\/pbs.twimg.com\/profile_banners\/71139907\/1506760383",
		"profile_link_color": "3D4B70",
		"profile_sidebar_border_color": "FFFFFF",
		"profile_sidebar_fill_color": "C9B280",
		"profile_text_color": "000000",
		"profile_use_background_image": true,
		"has_extended_profile": true,
		"default_profile": false,
		"default_profile_image": false,
		"following": true,
		"follow_request_sent": false,
		"notifications": false,
		"translator_type": "none"
	},
	"geo": null,
	"coordinates": null,
	"place": null,
	"contributors": null,
	"is_quote_status": false,
	"retweet_count": 0,
	"favorite_count": 1,
	"favorited": false,
	"retweeted": false,
	"possibly_sensitive": false,
	"possibly_sensitive_appealable": false,
	"lang": "en"
}
````

**Comment by annando** [comment]
Can you provide a screenshot from the post on your side? Connector posts are always private.

**Comment by MrPetovan** [comment]
![screen shot 2018-02-15 at 4 29 17 pm](https://user-images.githubusercontent.com/925415/36281973-91f04f56-126d-11e8-9fd6-ebe266947d4f.png)

**Comment by MrPetovan** [comment]
> Connector posts are always private.

It only makes sense for Twitter posts if the author has a protected status though.

**Comment by MrPetovan** [comment]
> It only makes sense for Twitter posts if the author has a protected status though.

Scratch that, we already set the post as private only if the account is protected, you should be able to see the post on my server. Did you even try? 😋

**Comment by annando** [comment]
The "private" field is set to "2" for connector posts if I remember that correctly.

**Comment by MrPetovan** [comment]
Not with the Twitter connector. 1 if remote user is protected, 0 otherwise.

**Comment by annando** [comment]
Now I know the difference: Twitter posts aren't mirrored with "uid=0".

**Comment by MrPetovan** [comment]
Shouldn’t they get a shadow item though?

**Comment by annando** [comment]
Until now we decided not to do so.

**Comment by MrPetovan** [comment]
How is it different from me receiving a Diaspora public message from one of my contacts?

**Comment by annando** [comment]
It came from a connector. By now only "native" posts are stored that way. And I guess this is good since we couldn't comment on these connector posts.

**Comment by MrPetovan** [comment]
Ah, I see the difference now. Does it mean you can't see the URL I pasted in the issue?

**Comment by annando** [comment]
Correct.

**Comment by MrPetovan** [comment]
Isn't there a flag to prevent commenting (writable = 0?) to stil have a shadow entry?

**Comment by annando** [comment]
With this writable it should be possible. But I guess it will be confusing for users that they can subscribe to tags and then they only can comment on some of them.

**Comment by MrPetovan** [comment]
Since it's explicitly from Twitter, I don't think it is that much of a surprise.

**Comment by MrPetovan** [comment]
After investigation, this is related to #2487, #2964 and #4451 as it concerns the autolinker feature. This time, it's the title of an attachment that contains a URL, and it's picked up by the autolinker, throwing the HTML out. This gives me yet another reason to pursue the lexer route.

**Comment by MrPetovan** [comment]
Hopefully fixed with #6835
